### PR TITLE
Added example to demonstrate differences between :class:`~.Transform` and :class:`~.ReplacementTransform`

### PR DESCRIPTION
**Pull request by vagrid**
## Changelog / Overview
Add documentation to the ReplacementTransform class with an example. 

DO NOT REMOVE THE FOLLOWING CHANGELOG LINES, EVEN IF YOU DON'T USE THEM.-->

## Motivation
suggestion:  Add more documentation to help (new) users.

## Explanation for Changes
I added some documentation for the ReplacementTransform class with an example showing the difference (to some extend) between Transform and ReplacementTransform

[Documentation Reference](https://manimce--####.org.readthedocs.build)

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/animation/transform.py [hunk 1]
```diff
@@ -144,6 +144,68 @@ def interpolate_submobject(
 
 
 class ReplacementTransform(Transform):
+    """Replace the mobject with target_mobject (like Transform does), and shows the target_mobject (whereas Transform shows the mobject)
+
+    Fades one mobject into another. Transform M1 is shown M2 doies not change, only M1
```
Comment (Nikhil0504) [inline manim/animation/transform.py:149]:
```suggestion
    Fades one mobject into another. Transform M1 is shown M2 does not change, only M1
```

**Review by Nikhil0504 — CHANGES_REQUESTED** [review]
Thank you for your contribution.

Overall the example looks good to me but do one change I mentioned.

File: manim/animation/transform.py [hunk 2]
```diff
@@ -144,6 +144,68 @@ def interpolate_submobject(
 
 
 class ReplacementTransform(Transform):
+    """Replace the mobject with target_mobject (like Transform does), and shows the target_mobject (whereas Transform shows the mobject)
```
Comment (hydrobeam) [inline manim/animation/transform.py:147]: The distinction between `ReplacementTransform` and `Transform` would be better suited as a note.
```suggestion
    """Replaces and morphs a mobject into the target mobject.
```

File: manim/animation/transform.py [hunk 3]
```diff
@@ -144,6 +144,68 @@ def interpolate_submobject(
...
+    Fades one mobject into another. Transform M1 is shown M2 doies not change, only M1
+
+        Parameters
+        ----------
+        mobject
+            The starting :class:`~.Mobject`.
+        target_mobject
+            The target :class:`~.Mobject`.
+        kwargs
+            Further keyword arguments are passed to the parent class.
```
Comment (hydrobeam) [inline manim/animation/transform.py:158]:
```suggestion
            Further keyword arguments that are passed to :class:`Transform`.
```

File: manim/animation/transform.py [hunk 4]
```diff
@@ -144,6 +144,68 @@ def interpolate_submobject(
...
+
+        Examples
+        --------
+
+        .. manim:: ReplacementTransformOrTransform
+
+            class ReplacementTransformOrTransform(Scene):
```
Comment (hydrobeam) [inline manim/animation/transform.py:165]: This example seems like a little excessive just to highlight the difference between the two animations. Plus, writing the latex and the animations would add a significant amount of time needed to build the docs. 

I think it could be be done neatly using squares.

**Review by hydrobeam — CHANGES_REQUESTED** [review]
A comment about the example and docs.

File: manim/animation/transform.py (hunk 4 above)
Comment (Nikhil0504) [reply manim/animation/transform.py:165]: I like the text example as users tend to get confused on the usage of this, but maybe we could have 3 text examples and 3 shapes?

Comment (hydrobeam) [reply manim/animation/transform.py:165]: Some squares with integers in them should achieve the same purpose and be much easier to understand. Another issue with this example is that it's very hard to parse because it's ~40 lines of code

Comment (vagrid) [reply manim/animation/transform.py]: Thank you both for your comments! I tried to take into account all of them in the modification. First contribution so it might not be good. Thanks again.

**Review by vagrid — COMMENTED** [review]
Modification of the documentation (based on comments), then example with integers instead of text (no additional explanation)

**Review by jsonvillanueva — COMMENTED** [review]
Though the example is fine, this uses too many wait/play calls resulting in a 17 second video that can easily be made 2 seconds. I've posted an example in #bot-testing that demonstrates how this example can be shortened: https://discord.com/channels/581738731934056449/780234936980078602/846844319423266877

File: manim/animation/transform.py [hunk 5]
```diff
@@ -144,6 +144,49 @@ def interpolate_submobject(
...
+                self.wait()
+                self.play(ReplacementTransform(integer_1, integer_2))
+                self.play(ReplacementTransform(integer_2, integer_3))
+                self.play(FadeOut(integer_3))
+                self.wait()
+                self.play(Transform(integer_4, integer_5))
+                self.play(Transform(integer_5, integer_6))
+                self.play(FadeOut(integer_6))
+                self.wait()
+    """
```
Comment (hydrobeam) [inline manim/animation/transform.py:188]: This is my take on organizing the example to use VGroups and arranges. Plus individual titles for each transform type.

```suggestion
    .. manim:: ReplacementTransformVsTransform

        class ReplacementTransformVsTransform(Scene):
            def construct(self):
                # set up the numbers
                r_transform = VGroup(*[Integer(i) for i in range(1,4)])
                text_1 = Text("ReplacementTransform", color=RED)
                r_transform.add(text_1)
                
                transform = VGroup(*[Integer(i) for i in range(4,7)])
                text_2 = Text("Transform", color=BLUE)
                transform.add(text_2)
        
                ints = VGroup(r_transform, transform)
                texts = VGroup(text_1, text_2).scale(0.75)
                r_transform.arrange(direction=UP, buff=1)
                transform.arrange(direction=UP, buff=1)
                
                ints.arrange(buff=2)        
                self.add(ints, texts)
        
                # The mobs replace each other and none are left behind
                self.play(ReplacementTransform(r_transform[0], r_transform[1]))
                self.play(ReplacementTransform(r_transform[1], r_transform[2]))
        
                # The mobs linger after the Transform()
                self.play(Transform(transform[0], transform[1]))
                self.play(Transform(transform[1], transform[2]))
                self.wait()
    """
```

Comment (vagrid) [reply manim/animation/transform.py:188]: Awesome!

**Review by hydrobeam — APPROVED** [review]
Looks good to me

**Review by Nikhil0504 — APPROVED** [review]
LGTM
